lib/ddtrace/notifications.rb

```diff
@@ -0,0 +1,34 @@
+module Datadog
+  class Notifications
+    def self.subscribe(pattern, tracer = Datadog.tracer, &block)
```

Comment:
I really like the elegance of this! Could it benefit from more customization? It might be useful to be able to pass options to the `trace` method (which would also remedy the `'<tmp>' span name.)

To do so, it could look something like:

```ruby
def self.subscribe(pattern, span_name, options = {}, tracer = Datadog.tracer, &block)
  subscriber = Subscriber.new(span_name, options, tracer, &block)
  ActiveSupport::Notifications.subscribe(pattern, subscriber)
end
```

Reply:
:+1: